File: 009-HarvardX-CS50/pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    // Getting input from user in dollars with validation
    float dollars;
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);
    
    // Converting dollars to cents
    int cents = round(dollars * 100);
    
    // Coins counter
    int quaters = floor(cents / 25);
    printf("%d\n", cents);
    cents = cents - 25 * quaters;
    int dimes = floor(cents / 10);
    cents = cents - 10 * dimes;
    int nickels = floor(cents / 5);
    cents = cents - 5 * nickels;
    int pennies = cents;
    int counter = quaters + dimes + nickels + pennies;
    printf("%d\n", counter);
}

File: 009-HarvardX-CS50/pset6/cash/cash.py
from cs50 import get_float
from math import floor

while True:
    dollars = get_float("Change owed: ")
    if dollars > 0:
        break

cents = round(dollars * 100)
quaters = floor(cents / 25)
cents = cents % 25
dimes = floor(cents / 10)
cents = cents % 10
nickels = floor(cents / 5)
cents = cents % 5
pennies = cents
print("{}".format(pennies + nickels + dimes + quaters))
